src/v/datalake/parquet_writer.h

```diff
@@ -18,7 +18,7 @@ namespace datalake {
 
 class arrow_to_iobuf {
 public:
-    explicit arrow_to_iobuf(const arrow::Schema& schema);
+    explicit arrow_to_iobuf(std::shared_ptr<arrow::Schema> schema);
```

Comment:
why are we changing the interface if we aren't extending the lifetime of the schema with the reference counted pointer?

Reply:
The arrow function that creates the schema returns a `shared_ptr`. The `batching_parquet_writer` passes that to this constructor.

Honestly, I'm not sure why I wrote this to take a reference instead of a shared_ptr before, the calls to it are all dereferencing the shared_ptr to get the reference.

Reply:
> Honestly, I'm not sure why I wrote this to take a reference instead of a shared_ptr before, the calls to it are all dereferencing the shared_ptr to get the reference.

one reason to take a reference is that the constructor doesn't hold on to the reference, so you are incrementing and then decrementing an atomic reference count for no reason. but doesn't really matter if this is going to be replaced by seastar-parquet.